tests/test-cases.js

```diff
@@ -585,7 +585,34 @@ var CORE_PARSER_TESTS = [
 			data: [['a', 'b', 'c'], ['']],
 			errors: []
 		}
-	}
+	},
+	{
+		description: "Simple duplicate header names",
+		input: 'A,A,A,A\n1,2,3,4',
+		config: { header: true },
+		expected: {
+			data: [['A', 'A_1', 'A_2', 'A_3'], ['1', '2', '3', '4']],
+			errors: []
+		}
+	},
+	{
+		description: "Duplicate header names where an original header matches intended substitution",
```

Comment:
I do not have a clear way that this is the best way to do. Probably we should respect the original names, but this makes code more complex.

As this is a corner case I think the best is to remove such test, and let the user decide what to do by using the transformFunction confiuguration.

Otherwise if we keep the test we are agreeing that this is the expected behaviour and I do not think is the right way.

Reply:
I agree. Have removed the test.